File: backend/app/models/payment.py
from sqlalchemy import Column, Integer, TIMESTAMP, ForeignKey, Enum
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.db.database import Base


class Payment(Base):
    __tablename__ = "payments"

    payment_id = Column(Integer, primary_key=True, index=True, autoincrement=True)
    order_id = Column(Integer, ForeignKey("orders.order_id"))
    payment_method = Column(Enum('cod', 'bank_transfer', 'credit_card', name='payment_method'), nullable=False)
    payment_status = Column(Enum('pending', 'paid', 'failed', name='payment_status'), default='pending')
    payment_date = Column(TIMESTAMP, server_default=func.current_timestamp())

    # Relationships
    order = relationship("Order", back_populates="payments")

File: api/MyAPI/MyAPI/Domain/Entities/Payment.cs

namespace MyAPI.Domain.Entities
{
    public class Payment
    {
        public int PaymentId { get; set; }

        public int OrderId { get; set; }

        public string PaymentMethod { get; set; } = "cod";

        public string PaymentStatus { get; set; } = "pending";

        public decimal Amount { get; set; }

        public string? TransactionCode { get; set; }

        public DateTime? PaidAt { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public Order Order { get; set; } = null!;
    }
}
